caracal/workers/transform_worker.py

```diff
@@ -71,152 +78,228 @@ def worker(pipeline, recipe, config):
+            if config['rewind_flags']['enable'] and label_in:
                 version = config['rewind_flags']['version']
                 if version in available_flagversions:
                     substep = 'rewind-{0:s}-ms{1:d}'.format(version, target_iter)
                     manflags.restore_cflags(pipeline, recipe, version, from_ms, cab_name=substep)
                     if available_flagversions[-1] != version:
                         substep = 'delete-flag_versions-after-{0:s}-ms{1:d}'.format(version, target_iter)
                         manflags.delete_cflags(pipeline, recipe,
-                            available_flagversions[available_flagversions.index(version)+1],
-                            from_ms, cab_name=substep)
+                                               available_flagversions[available_flagversions.index(version) + 1],
+                                               from_ms, cab_name=substep)
                 else:
                     manflags.conflict('rewind_to_non_existing', pipeline, wname, from_ms,
-                        config, flags_before_worker, flags_after_worker)
+                                      config, flags_before_worker, flags_after_worker)
 
             flagv = to_ms + '.flagversions'
+            tmp_ms = 'tmp_' + to_ms
+            tmpflagv = tmp_ms + '.flagversions'
             if pipeline.enable_task(config, 'split_field'):
                 msbase = os.path.splitext(to_ms)[0]
-                obsinfo_msname = to_ms
             else:
                 msbase = pipeline.msbasenames[i]
-                obsinfo_msname = from_ms
 
             summary_file = f'{msbase}-summary.json'
             obsinfo_file = f'{msbase}-obsinfo.txt'
 
             if pipeline.enable_task(config, 'split_field'):
                 step = 'split_field-ms{0:d}-{1:d}'.format(i, target_iter)
                 # If the output of this run of mstransform exists, delete it first
-                remove_output_products((to_ms, flagv, summary_file, obsinfo_file), directory=pipeline.msdir, log=log)
-
-                recipe.add('cab/casa_mstransform', step,
-                           {
-                               "vis": from_ms if label_in else from_ms + ":input",
-                               "outputvis": to_ms,
-                               "timeaverage": True if (config['split_field']['time_avg'] != '' and config['split_field']['time_avg'] != '0s') else False,
-                               "timebin": config['split_field']['time_avg'],
-                               "chanaverage": True if config['split_field']['chan_avg'] > 1 else False,
-                               "chanbin": config['split_field']['chan_avg'],
-                               "spw": config['split_field']['spw'],
-                               "datacolumn": dcol,
-                               "correlation": config['split_field']['correlation'],
-                               "scan": config['split_field']['scan'],
```

Comment:
`"scan": config['split_field']['scan']` is missing in the new mstransform below.

Reply:
@loi can you please fix this error spotted by @Athanaseus ?

Reply:
Thanks @francescaLoi